core/src/main/java/org/geysermc/geyser/translator/inventory/item/ItemTranslator.java

```diff
@@ -144,6 +148,99 @@ private static ItemData.Builder translateToBedrock(GeyserSession session, Item j
         return builder;
     }
 
+    private static CompoundTag addAttributes(CompoundTag nbt, Item item, String language) {
+        ListTag modifiers = nbt.get("AttributeModifiers");
+        if (modifiers == null) return nbt;
+        CompoundTag newNbt = nbt;
+        if (newNbt == null) {
+            newNbt = new CompoundTag("nbt");
+            CompoundTag display = new CompoundTag("display");
+            display.put(new ListTag("Lore"));
+            newNbt.put(display);
+        }
+        CompoundTag compoundTag = newNbt.get("display");
+        if (compoundTag == null) {
+            compoundTag = new CompoundTag("display");
+        }
+        ListTag listTag = compoundTag.get("Lore");
+
+        if (listTag == null) {
+            listTag = new ListTag("Lore");
+        }
+        String[] allSlots = new String[]{"mainhand", "offhand", "feet", "legs", "chest", "head"};
+        DecimalFormat decimalFormat = new DecimalFormat("0.#####");
+        Map<String, List<Tag>> slotsToModifiers = new HashMap<>();
+        for (String slot : allSlots) {
+            slotsToModifiers.put(slot, new ArrayList<>());
+        }
+        for (Tag modifier : modifiers) {
+            Map<String, Tag> modifierValue = (Map) modifier.getValue();
+            String[] slots = allSlots;
+            if (modifierValue.get("Slot") != null) {
+                slots = new String[]{(String) modifierValue.get("Slot").getValue()};
+            }
+            for (String slot : slots) {
+                List<Tag> list = slotsToModifiers.get(slot);
+                list.add(modifier);
+                slotsToModifiers.put(slot, list);
+            }
+        }
+
+        for (String slot : allSlots) {
+            List<Tag> modifiersList = slotsToModifiers.get(slot);
+            if (modifiersList.isEmpty()) continue;
+            Component slotComponent = Component.text()
+                    .resetStyle()
+                    .color(NamedTextColor.GRAY)
+                    .append(Component.newline(), Component.translatable("item.modifiers." + slot))
+                    .build();
+            listTag.add(new StringTag("", MessageTranslator.convertMessage(slotComponent, language)));
+
+
+            for (Tag modifier : modifiersList) {
+                Map<String, Tag> modifierValue = (Map) modifier.getValue();
+                double amount;
+                if (modifierValue.get("Amount") instanceof IntTag intTag) {
+                    amount = (double) intTag.getValue();
+                } else if (modifierValue.get("Amount") instanceof DoubleTag doubleTag) {
+                    amount = doubleTag.getValue();
+                } else {
+                    continue;
+                }
+                if (amount == 0) {
+                    continue;
+                }
+                int operation = (int) modifierValue.get("Operation").getValue();
```

Comment:
```suggestion
                ModifierOperation operation = ModifierOperation.from((int) modifierValue.get("Operation").getValue());
```
Would use the ModifierOperation enum so it's more clear what each number represents later in the code.

Reply:
I did this with my own commit because I needed to add import and change it later in the code.